crates/library/std/src/evm.fe

```diff
@@ -285,21 +285,36 @@ pub fn create2(value v: u256, buf: MemoryBuffer, salt s: u256) -> address {
     unsafe { return address(__create2(v, buf.offset(), buf.len(), s)) }
 }
 
-// TODO: return bool (success)
 pub fn call(gas: u256, addr: address, value: u256, mut buf: RawCallBuffer) -> bool {
-  unsafe{ return __call(gas, u256(addr), value, buf.offset(), buf.input_len(), buf.offset(), buf.output_len()) == 1 }
+    unsafe { return __call(gas, u256(addr), value, buf.offset(), buf.input_len(), buf.offset(), buf.output_len()) == 1 }
 }
 
-pub unsafe fn call_code(gas: u256, addr: address, value: u256, input_offset: u256, input_len: u256, output_offset: u256, output_len: u256) -> u256 {
-    return __callcode(gas, u256(addr), value, input_offset, input_len, output_offset, output_len)
+pub fn call_code(gas: u256, addr: address, value: u256, mut buf: RawCallBuffer) -> bool {
+    unsafe { return __callcode(gas, u256(addr), value, buf.offset(), buf.input_len(), buf.offset(), buf.output_len()) == 1 }
 }
 
-pub unsafe fn delegate_call(gas: u256, addr: address, value: u256, input_offset: u256, input_len: u256, output_offset: u256, output_len: u256) -> u256 {
-    return __delegatecall(gas, u256(addr), input_offset, input_len, output_offset, output_len)
+pub fn delegate_call(gas: u256, addr: address, mut buf: RawCallBuffer) -> bool {
+    unsafe { return  __delegatecall(gas, u256(addr), buf.offset(), buf.input_len(), buf.offset(), buf.output_len()) == 1 }
 }
 
-pub unsafe fn static_call(gas: u256, addr: address, input_offset: u256, input_len: u256, output_offset: u256, output_len: u256) -> u256 {
-    return __staticcall(gas, u256(addr), input_offset, input_len, output_offset, output_len)
+pub fn static_call(gas: u256, addr: address, mut buf: RawCallBuffer) -> bool {
+    unsafe { return  __staticcall(gas, u256(addr), buf.offset(), buf.input_len(), buf.offset(), buf.output_len()) == 1 }
+}
+
+pub unsafe fn call_2(gas: u256, addr: address, value: u256, input_offset: u256, input_len: u256, output_offset: u256, output_len: u256) -> bool {
```

Comment:
I guess these functions are made for scenarios where manual memory allocation is required (for example in precompiles.fe)